partitions.go

```diff
@@ -129,6 +131,26 @@ func (p *partitions) updateRemotePartitions(nodes []string) {
 		}
 	}
 
+	// Keep track of old peers in case zookeeper goes away.
+	for partitionId, partition := range remote {
+		newPartition := make([]string, len(partition))
+		copy(newPartition, partition)
+
+		unDedupedPartition := append(newPartition, p.old[partitionId]...)
+		found := map[string]bool{}
+
+		// Shitty dedupe, iterate though the remote peers
+		for _, node := range unDedupedPartition {
+			if !found[node] {
+				found[node] = true
+				p.old[partitionId] = append([]string{node}, p.old[partitionId]...)
```

Comment:
It'd probably be slightly more efficient to append rather than prepend, and then truncate as part of the dedupe (working backwards). Even better, though, would be to just have a separate `dedupe` method.